slm_porosity/heat_transfer.f90

```diff
@@ -256,14 +262,18 @@ SUBROUTINE user_algebraic_BC (Lu, u_in, nlocal, ne_local, jlev, meth)
         IF (ANY( face(1:dim) /= 0) .AND. ie == n_var_enthalpy) THEN
           CALL get_all_indices_by_face(face_type, jlev, nloc, iloc)
           IF (nloc > 0) THEN
-            IF (dim == 3 .AND. face(2) < 0) THEN
-              Lu(shift+iloc(1:nloc)) = du(ie, iloc(1:nloc), 2)
-            ELSEIF (face(dim) > 0) THEN
-              Lu(shift+iloc(1:nloc)) = &
-                Neumann_bc(enthalpy_prev(iloc(1:nloc)), psi_prev(iloc(1:nloc))) * du(ie, iloc(1:nloc), dim) + &
-                Dirichlet_bc(enthalpy_prev(iloc(1:nloc))) * u_in(shift+iloc(1:nloc))
+            IF (.NOT. adiabatic_BC) THEN
```

Comment:
If you write here `IF (adiabatic_BC) THEN`, your commit will be much shorter. Avoid multilevel `IF ... THEN` if possible. You can add this rule as a guideline.

Reply:
Moreover, your style is a potential source of conflicts during rebasing.

Reply:
I thought about that. It could be added two times more in already existing conditions. But I can't say which way is better.